rfc3164/message_test.go

```diff
@@ -62,3 +67,46 @@ func (s *Rfc3164MessageTestSuite) TestMessageCantParseMessage(c *C) {
   c.Assert("", Equals, msg.Pid())
 }
 
+/* Test recieving a mesaage from slightly in the past, over a year boundary */
+func (s *Rfc3164MessageTestSuite) TestMessageFromDecemberInJanuary(c *C) {
+  log := []byte("<94>Dec 29 20:07:15 webtest-mark simlogging[17155]: This is a log.info() message")
+  testDecDate := func() time.Time { return time.Date(2015, 1, 2, 0, 0, 0, 0, time.Now().Location()) } 
+
+  parser := NewParser(&log)
+  parser.TimeFunction = testDecDate
+  err := parser.Parse()
+  if nil != err {
+    c.Fatal(err)
+  }
+  var msg message.IMessage = parser.Message()
+  c.Assert(string(log), Equals, string(*msg.RawMessage()))
+  c.Assert("webtest-mark", Equals, msg.Hostname())
+  c.Assert("This is a log.info() message", Equals, msg.Message())
+  c.Assert(time.Date(2014, 12, 29, 20, 7, 15, 0, time.Now().Location()), Equals, msg.TimeStamp())
+  c.Assert("simlogging", Equals, msg.Process())
+  c.Assert(message.Info, Equals, msg.Severity())
+  c.Assert(message.Ftp, Equals, msg.Facility())
+  c.Assert("17155", Equals, msg.Pid())
```

Comment:
After a second pass, wouldn't it be more straight forward to call the fixTimestampIfNeeded() function directly in the tests?